src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -240,22 +240,14 @@ private void connect(boolean firstTrial) throws ConnectionTimeoutException {
     public void disconnect() throws ConnectionTimeoutException {
         log.info("Disconnecting...");
         state = State.Disconnecting;
-
         websocket.close();
-        waitForState(State.Disconnected);
```

Comment:
Slight change in semantics: method will no longer block caller until `waitForState` finishes.

Reply:
Never mind, reverted back to waiting.